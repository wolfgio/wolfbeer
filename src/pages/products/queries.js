import { gql } from 'apollo-boost';

export const SEARCH_CATEGORIES = gql`
  query allCategoriesSearch {
    allCategory{
      title
      id
    }
  }
`;

export const LIST_PRODUCTS = gql`
  query poc($id: ID!, $categoryId: Int, $search: String){
    poc(id: $id) {
      id
      products(categoryId: $categoryId, search: $search) {
        id
        title
        rgb
        images {
          url
        }
        productVariants {
          availableDate
          productVariantId
          price
          inventoryItemId
          shortDescription
          title
          published
          volume
          volumeUnit
          description
          subtitle
          components {
            id
            productVariantId
            productVariant {
              id
              title
              description
              shortDescription
            }
          }
        }
      }
    }
  }
`;
