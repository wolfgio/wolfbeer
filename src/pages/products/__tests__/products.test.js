import React from 'react';
import { create, act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import wait from 'waait';

import Products from '../products';
import { LIST_PRODUCTS, SEARCH_CATEGORIES } from '../queries';

const mocks = [
  {
    request: {
      query: LIST_PRODUCTS,
      variables: {
        id: '532',
        search: '',
        categoryId: null,
      },
    },
    result: {
      data: {
        poc: {
          id: '532',
          products: [
            {
              id: '8868',
              title: 'Skol 269ml - Unidade',
              rgb: false,
              images: [
                {
                  url: 'https://courier-images-codechallenge.s3-us-west-2.amazonaws.com/product/8868_205f958d-2e51-48a3-b4d5-a2998765571a.jpg',
                },
              ],
              productVariants: [
                {
                  availableDate: '2018-10-31T00:00:00',
                  productVariantId: '8502',
                  price: 2.09,
                  inventoryItemId: '80149',
                  shortDescription: null,
                  title: 'Skol 269ml - Unidade',
                  published: null,
                  volume: '0,00269',
                  volumeUnit: null,
                  description: null,
                  subtitle: 'Cervejas',
                  components: [],
                },
              ],
            },
          ],
        },
      },
    },
  },
  {
    request: {
      query: SEARCH_CATEGORIES,
      variables: {
        algorithm: 'NEAREST',
        lat: '-23.632919',
        long: '-46.699453',
        now: '2017-08-01T20:00:00.000Z',
      },
    },
    result: {
      data: {
        allCategory: [
          {
            title: 'Cervejas',
            id: '94',
          },
          {
            title: 'Destilados',
            id: '95',
          },
          {
            title: 'Vinhos',
            id: '92',
          },
          {
            title: 'Sem Ã¡lcool',
            id: '96',
          },
          {
            title: 'Petiscos',
            id: '97',
          },
          {
            title: 'Outros',
            id: '98',
          },
        ],
      },
    },
  },
];

describe('<Products />', () => {
  it('should render without error', async () => {
    let root;
    act(() => {
      root = create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Products />
        </MockedProvider>,
      );
    });

    await wait(300);

    expect(root.toJSON()).toMatchSnapshot();
  });
});
