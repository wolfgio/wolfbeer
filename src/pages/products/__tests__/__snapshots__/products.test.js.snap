// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> should render without error 1`] = `
<div
  className="sc-AykKD byAhix"
>
  <div
    className="sc-AykKE bxQStN"
  >
    <p>
      Entregar em:
    </p>
    <button
      className="sc-AykKF eHfyqP"
      onClick={[Function]}
    >
      <img
        className="sc-AykKG cHyhJT"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </button>
  </div>
  <h4
    style={
      Object {
        "margin": "16px 0",
      }
    }
  >
    Filtrar:
  </h4>
  <select
    className="sc-AykKJ ctTap"
    onChange={[Function]}
  >
    <option
      value="all"
    >
      Todas
    </option>
  </select>
  <input
    className="sc-AykKI hQrNgC"
    onChange={[Function]}
    placeholder="Pesquisar produto..."
    type="search"
  />
  <h4
    className="sc-AykKK eLluSA"
  >
    Items selecionados: 0
  </h4>
  <div
    className="sc-AykKH kvJHhB"
  >
    <h2>
      NÃ£o encontramos nenhum produto :/
    </h2>
  </div>
</div>
`;
